eth/backend.go

```diff
@@ -368,7 +368,9 @@ func (s *Ethereum) Protocols() []p2p.Protocol {
 // Start implements node.Lifecycle, starting all internal goroutines needed by the
 // Ethereum protocol implementation.
 func (s *Ethereum) Start() error {
-	s.setupDiscovery()
+	if err := s.setupDiscovery(); err != nil {
```

Comment:
Notably, Geth will stop immediately if an error occurs in the discovery setup, e.g. the provided DNS URL is invalid.

I think it should be the correct behavior. Any invalid DNS url should be detected as early as possible, instead of launching node silently.